src/xo-app/backup/restore/index.js

```diff
@@ -117,6 +152,19 @@ class _ModalBody extends Component {
         placeholder={intl.formatMessage(messages.importBackupModalSelectBackup)}
       />
       <br />
+      {vdis !== undefined && this.state.sr &&
+        <_Collapsible collapsible={vdis.length >= 3} buttonText='choose sr for each vdis '>
+          <br />
+          {map(vdis, vdi =>
+            <span>
+              <b>VDI:</b> {vdi.name}
+              <SelectSr key={vdi.uuid} onChange={this.linkState(`vdiSr.${vdi.uuid}`)} value={this._getSelectedValue(vdi.uuid)} predicate={sr => this._srPredicate(sr)} />
```

Comment:
`this._getSelectedValue` will be called on every render of this component. You can use the state to store the value instead. If you need to check that a value (in the props or the state) hasn't changed on each render, you can use a selector.